x-pack/plugins/cases/public/components/edit_connector/index.tsx

```diff
@@ -290,36 +242,47 @@ export const EditConnector = React.memo(
             <EuiFlexItem grow={false} data-test-subj="connector-edit-header">
               <h4>{i18n.CONNECTORS}</h4>
             </EuiFlexItem>
-            {isLoading && <EuiLoadingSpinner data-test-subj="connector-loading" />}
-            {!isLoading && !editConnector && permissions.push && actionsReadCapabilities && (
-              <EuiFlexItem data-test-subj="connector-edit" grow={false}>
-                <EuiButtonIcon
-                  data-test-subj="connector-edit-button"
-                  aria-label={i18n.EDIT_CONNECTOR_ARIA}
-                  iconType={'pencil'}
-                  onClick={onEditClick}
+            <EuiFlexItem data-test-subj="connector-edit" grow={false}>
+              <EuiButtonIcon
+                data-test-subj="connector-edit-button"
+                aria-label={i18n.EDIT_CONNECTOR_ARIA}
+                iconType="pencil"
+                onClick={onEditClick}
+                disabled={
+                  isLoading || editConnector || !hasPushPermissions || !actionsReadCapabilities
```

Comment:
Equals to `!isLoading && !editConnector && permissions.push && actionsReadCapabilities`